packages/mwp-app-render/src/components/GoogleTagManager.jsx

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// Runtime flag
+export const GOOGLE_TAG_MANAGER_RTF = 'google_tag_manager';
+
+// Production and dev keys
+const GTM_KEYS = {
+	'prod': 'GTM-T2LNGD', 
+	'dev': 'GTM-W9W847',
+};
+
+/*
+ * @description Gets google tag manager script with key applied
+ * @see {@link https://developers.google.com/tag-manager/quickstart}
+*/
+const getScript = (key: string) => {
+	const script = `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);})(window,document,'script','dataLayer','${key}');` 
+	return <script>{script}</script>
+};
+
+/*
+ * @description Gets google tag manager noscript with key applied
+ * @see {@link https://developers.google.com/tag-manager/quickstart}
+*/
+const getNoscript = (key: string) => { 
+	const src = `https://www.googletagmanager.com/ns.html?id=${key}`;
+	return <noscript><iframe src={src} height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
+};
+
+/*
+ * @description Gets google tag manager tags (script or noscript)
+ * @param {String} tag 'script' or 'noscript', indicating which GTM tag to return
+ * @param {Boolean} rtf Whether or not the runtime flag is on or off
+ * @param {Boolean} isProd Whether or not environment is production
+ * @see {@link https://developers.google.com/tag-manager/quickstart}
+ *
+ * @example
+ * <head>
+ * 	<!-- GTM script -->
+ * 	<GoogleTagManager tag="script" rtf={true} isProd={true} />
+ * </head>
+ * <body>
+ * 	<!-- GTM noscript -->
+ * 	<GoogleTagManager tag="noscript" rtf={true} isProd={true} />
+ * </body>
+*/
+const GoogleTagManager = ({tag, rtf = false, isProd = false}) => {
+
+	const validTagProp = ['script','noscript'].indexOf(tag) >= 0;
+	
+	if(!validTagProp || !rtf) {
```

Comment:
why pass in `rtf`? Since the parent component is responsible for determining the RTF state, it seems reasonable to let it also decide what to do with it? i.e. don't try to render `<GoogleTagManager />` at all if the rtf if false?

Reply:
Yeah, I had considered both approaches.

I like something like `{gtmRtf && <GoogleTagManager />}` as well, because it doesn’t try to render the component if the flag is `false`.

But my thinking behind passing `rtf` instead was something like…
- Keeps the runtime flag controlled rendering behavior with the component
- Testable in `GoogleTagManager.test.jsx`
- Hopefully aids in making it clear that a rtf exists and is necessary to render, just by looking <GoogleTagManager /> docs
- It allows the component to not render by default (which I preferred in this case since it’s a 3rd party script) and requires the implementer to explicitly pass a `true` rtf to get it to render. Otherwise, though probably not likely, checking the runtime flag could be more easily forgotten.

Reply:
```js
const GTM_KEY = process.env.NODE_ENV === 'production' ? 'foo': 'bar'
const GoogleTagManager = ({tag, rtf = false, key = GTM_KEY}) => {
  ...
}
```